src/arfsdao.ts

```diff
@@ -152,57 +154,118 @@ export class ArFSDAOAnonymous extends ArFSDAOType {
-			const txData = await this.arweave.transactions.getData(driveBuilder.txId, { decode: true });
-			const dataString = await Utf8ArrayToStr(txData);
-			const dataJSON = await JSON.parse(dataString);
+		const { data } = response.data;
+		const { transactions } = data;
+		const { edges } = transactions;
 
-			// Get the drive name and root folder id
-			driveBuilder.name = dataJSON.name;
-			driveBuilder.rootFolderId = dataJSON.rootFolderId;
+		if (!edges.length) {
+			throw new Error(`Public folder with Folder ID ${folderId} not found!`);
 		}
 
-		return driveBuilder.build();
+		const folderBuilder = new ArFSPublicFolderBuilder(folderId, this.arweave);
+		return await folderBuilder.build();
+	}
+
+	async getPublicFilesWithParentFolderIds(
+		folderIDs: FolderID[],
+		latestRevisionsOnly = false
+	): Promise<ArFSPublicFile[]> {
+		let cursor = '';
+		let hasNextPage = true;
+		const allFiles: ArFSPublicFile[] = [];
+
+		while (hasNextPage) {
+			const gqlQuery = buildQuery(
+				[
+					{ name: 'Parent-Folder-Id', value: folderIDs },
+					{ name: 'Entity-Type', value: 'file' }
+				],
+				cursor
+			);
+
+			const response = await this.arweave.api.post(graphQLURL, gqlQuery);
+			const { data } = response.data;
+			const { transactions } = data;
+			const { edges } = transactions;
+			hasNextPage = transactions.pageInfo.hasNextPage;
+			const files: Promise<ArFSPublicFile>[] = edges.map(async (edge: GQLEdgeInterface) => {
+				const { node } = edge;
+				cursor = edge.cursor;
+				const fileBuilder = await ArFSPublicFileBuilder.fromArweaveNode(node, this.arweave);
+				return fileBuilder.build(node);
+			});
+			allFiles.push(...(await Promise.all(files)));
+		}
+		return latestRevisionsOnly ? allFiles.filter(latestRevisionFilter) : allFiles;
+	}
+
+	async getAllFoldersOfPublicDrive(driveId: DriveID, latestRevisionsOnly = false): Promise<ArFSPublicFolder[]> {
+		let cursor = '';
+		let hasNextPage = true;
+		const allFolders: ArFSPublicFolder[] = [];
+
+		while (hasNextPage) {
+			const gqlQuery = buildQuery(
+				[
+					{ name: 'Drive-Id', value: driveId },
+					{ name: 'Entity-Type', value: 'folder' }
+				],
+				cursor
+			);
+
+			const response = await this.arweave.api.post(graphQLURL, gqlQuery);
+			const { data } = response.data;
+			const { transactions } = data;
+			const { edges } = transactions;
+			hasNextPage = transactions.pageInfo.hasNextPage;
+			const folders: Promise<ArFSPublicFolder>[] = edges.map(async (edge: GQLEdgeInterface) => {
+				const { node } = edge;
+				cursor = edge.cursor;
+				const folderBuilder = ArFSPublicFolderBuilder.fromArweaveNode(node, this.arweave);
+				return await folderBuilder.build(node);
+			});
+			allFolders.push(...(await Promise.all(folders)));
+		}
+		return latestRevisionsOnly ? allFolders.filter(latestRevisionFilter) : allFolders;
+	}
+
+	/**
+	 * Lists the children and self of certain public folder
+	 * @param {FolderID} folderId the folder ID to list children of
```

Comment:
Docblock needs updates

Reply:
614fe0b796d24b994e18927fb971ae683c9f8ccf

Reply:
It's missing the explanation about max-depth!